Sources/TensorFlow/Operators/Math.swift

```diff
@@ -1895,11 +1969,22 @@ internal extension Tensor where Scalar: TensorFlowFloatingPoint {
     func _vjpMean(squeezingAxes axes: Tensor<Int32>) -> (Tensor, (Tensor) -> Tensor) {
         let value = mean(squeezingAxes: axes)
         let count = Raw.gather(params: shapeTensor, indices: axes).product()
-        return (value, { [shape = shapeTensor] v in
-	      let unsqueezed = v.expandingShape(at: axes.scalars.map { Int($0) })
+        return (value, { [shape = shapeTensor] seed in
+	      let unsqueezed = seed.expandingShape(at: axes.scalars.map { Int($0) })
 	      return unsqueezed.broadcasted(toShape: shape) / Tensor(count)
 	    })
     }
+
+    @inlinable
+    func _vjpCumulativeSum(
+        alongAxis axis: Tensor<Int32>,
```

Comment:
Shouldn’t this be “axes” instead, since it’s a tensor?

Reply:
Cumulative sum can only be performed over a single axis. Since we cannot enforce that in the tensor type, I hope the name make it clear. I can also add a precondition although that would cause lazy tensor materialization and so I avoided it.

Reply:
Why not take an Int instead?

Reply:
To allow it to be used without lazy tensor materialization in cases where that axis is the result of some other tensor operation (e.g., `Tensor.rank` or `Tensor.shape[i]`). I have also added an overload taking `Int` directly.

Reply:
Oh ok!

Reply:
Tests pass so if you approve this, I can go ahead and merge it.